platform/android/src/style/formatted.cpp

```diff
@@ -0,0 +1,42 @@
+#include "formatted.hpp"
+#include "formatted_section.hpp"
+
+namespace mbgl {
+namespace android {
+
+void Formatted::registerNative(jni::JNIEnv& env) {
+    jni::Class<Formatted>::Singleton(env);
+}
+
+jni::Local<jni::Object<Formatted>> Formatted::New(jni::JNIEnv& env, const style::expression::Formatted& value) {
+    static auto& formatted = jni::Class<Formatted>::Singleton(env);
+    static auto formattedConstructor = formatted.GetConstructor<jni::Array<jni::Object<FormattedSection>>>(env);
+    static auto& formattedSection = jni::Class<FormattedSection>::Singleton(env);
+    static auto formattedSectionConstructor = formattedSection.GetConstructor<jni::String, double, jni::Array<jni::String>>(env);
+
+    auto sections = jni::Array<jni::Object<FormattedSection>>::New(env, value.sections.size());
+    for (std::size_t i = 0; i < value.sections.size(); i++) {
+        auto section = value.sections.at(i);
+        auto text = jni::Make<jni::String>(env, section.text);
+
+        auto fontScale = 1.0;
+        if (section.fontScale) {
+            fontScale = section.fontScale.value();
+        }
+
+        auto fontStack = jni::Array<jni::String>::New(env, 0);
```

Comment:
When I saw `FormattedSection::getFontStack()` above and saw that it was nullable, I assumed that "null" was being used to represent "not set". But it looks here (and now I notice in the tests) that `text-font: null` will cause `getFontStack()` to return `new String[] {}`. That doesn't seem right -- an empty array would signify "use this FontStack that doesn't have any fonts in it" instead of "use the default font stack".

Reply:
Good point, I'm going to return `null` instead of an empty array or un-set stack.